homeassistant/components/plugwise/config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Config flow for Plugwise integration."""
+import logging
+
+from Plugwise_Smile.Smile import Smile
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_HOST, CONF_PASSWORD
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_HOST): str, vol.Required(CONF_PASSWORD): str}
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """
+    Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    websession = async_get_clientsession(hass, verify_ssl=False)
+    api = Smile(
+        host=data["host"], password=data["password"], timeout=30, websession=websession
+    )
+
+    try:
+        await api.connect()
+    except Smile.InvalidAuthentication:
+        raise InvalidAuth
+    except Smile.ConnectionFailedError:
+        raise CannotConnect
+
+    return {"title": api.smile_name}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Plugwise Smile."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+
```

Comment:
Does the device have a unique identifier so we can make sure the user doesn't set it up twice?

```
await self.async_set_unique_id(serial)
self._abort_if_unique_id_configured()
```

Reply:
The (plugwise) ID of the gateway and/or it's IP-address as it is a local-only device?

Reply:
The unique id needs to be stable for the life of the config entry.  We shouldn't use an IP address as that can change.

Good sources for a unique ID:

- Serial number of a device
- MAC address of a device
- latitude/longitude

Reply:
Yup, understood, I prefer IDs over changeable IPs as well, adding that.

Reply:
Included now